libsplinter/src/admin/service/shared.rs

```diff
@@ -913,6 +1062,69 @@ impl AdminServiceShared {
         Ok(())
     }
 
+    fn check_connected_peers_payload_disband(
+        &mut self,
+        members: &[SplinterNode],
+        payload: CircuitManagementPayload,
+        message_sender: String,
+    ) -> Result<(), ServiceError> {
+        let mut missing_protocol_ids = vec![];
+        let mut pending_peers = vec![];
+        let mut pending_members = vec![];
+        let mut added_peers: Vec<String> = vec![];
+        for node in members {
+            if self.node_id() != node.get_node_id() {
+                debug!("Referencing node {:?}", node);
+                let peer_ref = self
```

Comment:
This should not be adding peer_refs, required peer_refs will have been added when the circuit was created. This is adding an extra set of peer_refs that will not be removed when the disband request is approved. Only create_payloads should add new peer_refs